ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala

```diff
@@ -395,6 +399,33 @@ abstract class EventStorageBackendTemplate(
     "NULL as driver_metadata",
   ).mkString(", ")
 
+  val EventSequentailIdFromTo: RowParser[(Long, Long)] =
+    long("event_sequential_id_first") ~ long("event_sequential_id_last") map {
+      case event_sequential_id_first ~ event_sequential_id_last =>
+        (event_sequential_id_first, event_sequential_id_last)
+    }
+
+}
+
+abstract class EventStorageBackendTemplate(
+    eventStrategy: EventStrategy,
+    queryStrategy: QueryStrategy,
+    ledgerEndCache: LedgerEndCache,
+    stringInterning: StringInterning,
+    // This method is needed in pruneEvents, but belongs to [[ParameterStorageBackend]].
+    participantAllDivulgedContractsPrunedUpToInclusive: Connection => Option[Offset],
+) extends EventStorageBackend {
+  import com.daml.platform.store.backend.Conversions.ArrayColumnToIntArray._
+  import EventStorageBackendTemplate._
+
+  private val logger: ContextualizedLogger = ContextualizedLogger.get(this.getClass)
+
+  override def transactionStreamingQueries: TransactionStreamingQueries =
```

Comment:
The refactoring that introduces this new `TransactionStreamingQueries` class makes the job of the reviewer very difficult indeed. Would it be possible to split a deep refactoring like this one and a functional change separate?

Reply:
Discussed offline, these queries are almost all new and so it's ok to leave them in `TransactionStreamingQueries` class